demonstrations/tutorial_shadow_hamiltonian_simulation.py

```diff
@@ -0,0 +1,321 @@
+
+.. math:: \frac{\text{d}}{\text{dt}} |\rho\rangle = - i H_S |\rho\rangle.
+
+The Hamiltonian matrix :math:`H_S` is given by the commutation relations between the system Hamiltonian :math:`H` and
+the operators in :math:`S = \{O_m\}`. Namely, it is implicitly defined by
+
+.. math:: [H, O_m] = - \sum_{m'=1}^M \left( H_S \right)_{m m'} O_{m'}.
+
+Let us solve for the matrix elements :math:`(H_S)_{m m'}$.  To do this, recall that a vector :math:`\boldsymbol{v}` can always be decomposed in an orthogonal basis :math:`\boldsymbol{e}_j` via
+:math:`\boldsymbol{v} = \sum_j \frac{\langle \boldsymbol{e}_j, \boldsymbol{v}\rangle}{||\boldsymbol{e}_j||^2} \boldsymbol{e}_j`.
+Since the operators under consideration are elements of the vector space of Hermitian operators, we can use that to compute :math:`H_S`.
+
+In particular, with the trace inner product, this amounts to
+
+.. math:: [H, O_m] = \sum_{m'=1}^M \frac{\text{tr}\left( O_{m'} [H, O_m] \right)}{|| O_{m'} ||^2} O_{m'},
+
+from which we can read off the matrix elements of :math:`H_S`, i.e.,
+
+.. math:: (H_S)_{m m'} = -\frac{\text{tr}\left( O_{m'} [H, O_m] \right)}{|| O_{m'} ||^2}.
+
+From this we can readily see that the operators :math:`O_m` need to be chosen such that they support :math:`[H, O_m]`.
+
+How this relates to g-sim
+-------------------------
+
+In :doc:`g-sim </demos/tutorial_liesim>`
+[#Somma]_ [#Somma2]_ [#Galitski]_ [#Goh]_, we have operators :math:`\{ g_i \}` that are potential generators or observables for a parametrized quantum circuit,
+e.g. :math:`U(\theta) = \prod_\ell \exp(-i \theta_\ell g_\ell)`.
+For that, we are looking at the so-called dynamical Lie algebra (DLA)
+:math:`\mathfrak{g} = \langle \{ g_i \} \rangle_\text{Lie} = \{ g_1, .., g_{|\mathfrak{g}|} \}` of the circuit as well as
+the adjoint representation
+:math:`(-i \text{ad}_{g_\gamma})_{\alpha \beta} = f^\gamma_{\alpha \beta}`, where :math:`f^\gamma_{\alpha \beta}` are the 
+:func:`~pennylane.structure_constants` of the DLA.
+In g-sim, we also evolve expectation vectors :math:`(\vec{g})_i = \langle g_i \rangle`.
+In particular, the circuit of evolving a state according to :math:`U(\theta)` and computing expectation values 
+:math:`\langle g_i \rangle` then corresponds to evolving :math:`\vec{g}` by :math:`\prod_\ell \exp(-i \theta_\ell \text{ad}_{g_\ell})`.
+See :doc:`our demo on g-sim </demos/tutorial_liesim>` for further details.
+
+Shadow Hamiltonian simulation can thus be viewed as g-sim
+with a single, specific gate :math:`U(\theta) = e^{-i \theta H}` and parameter :math:`\theta = t`, and run on a quantum computer.
+
+In particular, :math:`H_S` corresponds to the adjoint representation :math:`\text{ad}_H` of the 
+:doc:`dynamical Lie algebra </demos/tutorial_liealgebra>`
+:math:`\langle \{ H \} \cup S \rangle_\text{Lie}`. We explain the concept of the adjoint representation in our
+:doc:`demo on g-sim </demos/tutorial_liesim>` that makes extensive use of it.
```

Comment:
Note that we do not need to close the set S under commutators, so it is not really the adjoint rep of an algebra, right?

Reply:
Ah yes, you also say this right below :sweat_smile: So maybe
```suggestion
In particular, :math:`H_S` corresponds to a part of the adjoint representation :math:`\text{ad}_H` of the 
:doc:`dynamical Lie algebra </demos/tutorial_liealgebra>`
:math:`\langle \{ H \} \cup S \rangle_\text{Lie}`. We explain the concept of the adjoint representation in our
:doc:`demo on g-sim </demos/tutorial_liesim>` that makes extensive use of it.
```

Reply:
> so it is not really the adjoint rep of an algebra, right?

Not quite, but the mathematical definition is the same. I didnt want to get into calling this a pseudo-adjoint repr and just figured it'd be fine to make the general connection to the adjoint repr and then specify that the full thing is not needed and we can just use a subset of that 🤔

Reply:
I understand. I think it might be nice to leave whichever small indication that the full rep is not needed at this point already, in order to avoid
- confusion
- people stopping here, thinking that things are not efficient for large DLAs
:thinking:

Reply:
yeah good point! will incorporate!